socorro/processor/breakpad_transform_rules.py

```diff
@@ -459,3 +491,153 @@ def _predicate(
         return 'create_dump_lookup' in raw_crash
 
 
+#==============================================================================
+class BreakpadStackwalkerRule2015(ExternalProcessRule):
+
+    required_config = Namespace()
+    required_config.add_option(
+        name='symbols_url',
+        doc='url of the symbol server',
+        default="https://localhost"
+    )
+    required_config.command_line = change_default(
+        ExternalProcessRule,
+        'command_line',
+        'timeout -s KILL 30 {command_pathname} '
+        '--raw-json {raw_crash_pathname} '
+        '--symbols-url {symbols_url} '
+        '--symbols-cache {symbol_cache_path} '
+        '{dump_file_pathname} '
+        '2>/dev/null'
+    )
+    required_config.command_pathname = change_default(
+        ExternalProcessRule,
+        'command_pathname',
+        '/data/socorro/stackwalk/bin/stackwalker',
+    )
+    required_config.add_option(
+        'symbol_cache_path',
+        doc='the path where the symbol cache is found, this location must be '
+        'readable and writeable (quote path with embedded spaces)',
+        default='%s/symbols' % tempfile.gettempdir(),
+    )
+    required_config.add_option(
+        'temporary_file_system_storage_path',
+        doc='a path where temporary files may be written',
+        default=tempfile.gettempdir(),
+    )
+
+
+    #--------------------------------------------------------------------------
+    #def __init__(self, config):
+        #super(BreakpadStackwalkerRule, self).__init__(config)
+
+
+    #--------------------------------------------------------------------------
+    def version(self):
+        return '1.0'
+
+    #--------------------------------------------------------------------------
+    @contextmanager
+    def _temp_raw_crash_json_file(self, raw_crash, crash_id):
+        file_pathname = os.path.join(
+            self.config.temporary_file_system_storage_path,
+            "%s.%s.TEMPORARY.json" % (
+                crash_id,
+                threading.currentThread().getName()
+            )
+        )
+        with open(file_pathname, "w") as f:
+            ujson.dump(dict(raw_crash), f)
```

Comment:
the `dict(some_mapping)` will create a duplicate in memory and considering that these raw crashes are sometimes very large that can mean a lot of memory.